inspire-atom/src/main/java/org/fao/geonet/services/inspireatom/AtomSearch.java

```diff
@@ -142,9 +144,11 @@ public Element feeds(
             // Retrieve the datasets related to the service metadata
             datasetIdentifiers = InspireAtomUtil.extractRelatedDatasetsIdentifiers(schema, md, dm);
 
-            // Add query filter / TODO Migrate ?
-            //            String values = Joiner.on(" or ").join(datasetIdentifiers);
-            //            params.addContent(new Element("identifier").setText(values));
+            String datasets = datasetIdentifiers.stream().collect(Collectors.joining("\",\"", "\"", "\""));
```

Comment:
Escape the identifiers since they can contain any character and can create a malformed JSON string later.

```suggestion
            String datasets = datasetIdentifiers.stream().map(StringEscapeUtils::escapeJson)
                .collect(Collectors.joining("\",\"", "\"", "\""));
```